>Created connection pool, using Simple Implementation in DatabaseCont…

ivaylomitev17 opened the pull request:
>Created Connection Pool, realising Simple Implementation in DatabaseControlUnit.java
>Created interface Database Manipulation <T>, implemented by all the services
>Created services for operating with "order" and "user" table
>Created new model Order.java
>UserVerification deleted and its methods integrated in UserService.java
>Created custom exception, thrown when new price of product is set to be lower than previously set minimum price

ivaylomitev17 committed:
>Created connection pool, using Simple Implementation in DatabaseControlUnit.java
>Created interface Database Manipulation <T>, implemented by all the services
>Created services for operating with "order" and "user" table
>Created new model Order.java
>UserVerification deleted and its methods integrated in UserService.java
>Created custom exception, thrown when new price of product is set to be lower than previously set minimum price

ivaylomitev17 committed:
Delete ProjectSAP1.iml

dzhurdzhina commented on src/tu/sofia/shop/black/friday/Exceptions/NewPriceLowerThanMinimumException.java:
```diff
@@ -0,0 +1,4 @@
+package tu.sofia.shop.black.friday.Exceptions;
+
+public class NewPriceLowerThanMinimumException extends Exception {
+}
```
An empty class?

dzhurdzhina commented on src/tu/sofia/shop/black/friday/ServerMain.java:
```diff
@@ -1,21 +1,10 @@
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+        DatabaseControlUnit databaseControl = new DatabaseControlUnit();
```
It does nothing?

dzhurdzhina commented on src/tu/sofia/shop/black/friday/model/Order.java:
```diff
@@ -0,0 +1,25 @@
+
+public class Order {
+    private int buyerId;
+    private Date date;
```
I suggest that information about products together with quantity and associated properties (size, color, etc.) shall be kept here as well as price

dzhurdzhina commented on src/tu/sofia/shop/black/friday/util/DatabaseControlUnit.java:
```diff
@@ -2,30 +2,50 @@
-    private static Statement statement;
+    public DatabaseControlUnit (){
+        List<Connection> pool = new ArrayList<>(MAX_CONNECTIONS);
+        for (int i = 0; i < MAX_CONNECTIONS; i++) {
```
Loops, try-catch statements, etc. are bad practice within constructor. The constructor purpose is only to initialize class fields and nothing else.

dzhurdzhina commented on src/tu/sofia/shop/black/friday/util/DatabaseManipulation.java:
```diff
@@ -1,21 +1,13 @@
+    void updateDatabase(String column, String value, String  condition) throws SQLException;
+    void insertIntoDatabase(T o) throws SQLException;
+    ResultSet selectFromDatabase(String whereCondition) throws SQLException;
+    void setConnection(Connection connection);
```
Connection establishment is not a manipulation within a database, it cannot be declared within such an interface.

ivaylomitev17 committed:
>added controllers to operate with the services
>added some new exceptions
>added thread connection between client and server
>added ClientHandler.java, operating as router between the client messages and the server`s services
